From 6a1a0927491121ef090a7e573057828ed2007692 Mon Sep 17 00:00:00 2001
From: Tao Huang <huangtao@rock-chips.com>
Date: Mon, 4 Sep 2017 14:34:30 +0800
Subject: [PATCH] soc: rockchip: add android version support

Add config ANDROID_VERSION, default is Android 7.1.
And add ANDROID_VERSION macro.

Change-Id: I6309ea9118c1d847224f329f6c2d6ab7e073ee5a
Signed-off-by: Tao Huang <huangtao@rock-chips.com>
---
 drivers/soc/rockchip/Kconfig           |  8 ++++++++
 include/soc/rockchip/android-version.h | 17 +++++++++++++++++
 2 files changed, 25 insertions(+)
 create mode 100644 include/soc/rockchip/android-version.h

diff --git a/drivers/soc/rockchip/Kconfig b/drivers/soc/rockchip/Kconfig
index aacbd43613b2..0a800e78ced4 100644
--- a/drivers/soc/rockchip/Kconfig
+++ b/drivers/soc/rockchip/Kconfig
@@ -3,6 +3,14 @@ if ARCH_ROCKCHIP || COMPILE_TEST
 #
 # Rockchip Soc drivers
 #
+config ANDROID_VERSION
+	hex "Android Version"
+	depends on ANDROID
+	default "0x07010000"
+	help
+	  The version of Android kernel supported.
+	  The value is 0xAABBCCRR, corresponding to Android A.B.C release R
+
 config ROCKCHIP_CPUINFO
 	bool "Rockchip cpuinfo support"
 	depends on ROCKCHIP_EFUSE && (ARM64 || ARM)
diff --git a/include/soc/rockchip/android-version.h b/include/soc/rockchip/android-version.h
new file mode 100644
index 000000000000..e0d865bab335
--- /dev/null
+++ b/include/soc/rockchip/android-version.h
@@ -0,0 +1,17 @@
+/*
+ * Copyright (C) 2017 Rockchip Electronics Co. Ltd.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of version 2 of the GNU General Public License as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ */
+
+#ifndef ANDROID_VERSION
+#define ANDROID_VERSION(a, b, c, r) \
+	(((a) << 24) + ((b) << 16) + ((c) << 8) + (r))
+#endif
-- 
2.35.3

